src/app/calculator/furnaces/air-heating/air-heating-form/air-heating-form.component.ts

```diff
@@ -0,0 +1,275 @@
+  @HostListener('window:resize', ['$event'])
+  onResize(event) {
+    this.setOpHoursModalWidth();
+  }
+  form: FormGroup;
+  fuelOptions: any;
+
+  resetDataSub: Subscription;
+  generateExampleSub: Subscription;
+
+  calculationMethods: Array<string> = [
+    'Excess Air',
+    'Oxygen in Flue Gas'
+  ];
+
+  calculationExcessAir: number = 0.0;
+  calculationFlueGasO2: number = 0.0;
+  calcMethodExcessAir: boolean;
+  formWidth: number;
+
+  showOperatingHoursModal: boolean;
+  warnings: FlueGasWarnings;
+  o2Warning: string;
+
+  constructor(private suiteDbService: SuiteDbService,
+              private airHeatingService: AirHeatingService, 
+              private airHeatingFormService: AirHeatingFormService,
+              private phastService: PhastService) { }
+
+  ngOnInit() {
+    this.initSubscriptions();
+  }
+
+  initSubscriptions() {
+    this.resetDataSub = this.airHeatingService.resetData.subscribe(value => {
+      this.initForm();
+    })
+    this.generateExampleSub = this.airHeatingService.generateExample.subscribe(value => {
+      this.initForm();
+    })
+  }
+
+  ngAfterViewInit() {
+    setTimeout(() => {
+      this.setOpHoursModalWidth();
+    }, 100)
+  }
+
+  ngOnDestroy() {
+    this.resetDataSub.unsubscribe();
+    this.generateExampleSub.unsubscribe();
+  }
+
+  initForm() {
+    let airHeatingInput: AirHeatingInput = this.airHeatingService.airHeatingInput.getValue();
+    this.form = this.airHeatingFormService.getAirHeatingForm(airHeatingInput);
+    this.initFormSetup();
+  }
+
+  initFormSetup() {
+    this.setFuelOptions();
+    if (this.form.controls.materialTypeId.value && this.form.controls.materialTypeId.value !== '') {
+      if (this.form.controls.CH4.value === '' || !this.form.controls.CH4.value) {
+        this.setMaterialProperties();
+      }
+    }
+    this.setCalcMethod();
+    this.calcExcessAir();
+  }
+
+  changeFuelType() { {}
+    let currentInput: AirHeatingInput;
+    if (this.form.controls.gasFuelType.value == true) {
+      currentInput = this.airHeatingFormService.getAirHeatingInputGasMaterial(this.form);
+    } else {
+      currentInput = this.airHeatingFormService.getAirHeatingInputSolidMaterial(this.form);
+    }
+    this.airHeatingService.airHeatingInput.next(currentInput);
+    this.initForm();
+  }
+
+  setFuelOptions() {
+    if (this.form.controls.gasFuelType.value == true) {
+      this.fuelOptions = this.suiteDbService.selectGasFlueGasMaterials();
+    } else {
+      this.fuelOptions = this.suiteDbService.selectSolidLiquidFlueGasMaterials();
+    }
+  }
+
+  calcExcessAir() {
+    let input = this.airHeatingFormService.getMaterialInputProperties(this.form);
+
+    if (!this.calcMethodExcessAir) {
+      if (this.form.controls.flueGasO2.status === 'VALID') {
+        if (this.form.controls.gasFuelType.value == true) {
+          this.calculationExcessAir = this.phastService.flueGasCalculateExcessAir(input);
```

Comment:
looks like these functions are taking `any` as the parameter type... Investigate if thats a quick fix or should use a new issue please.